lib/notifications/notification.ex

```diff
@@ -27,6 +53,29 @@ defmodule Notifications.Notification do
     :operator_id,
     :operator_name,
     :route_id_at_creation,
-    :start_time
+    :start_time,
+    :end_time
   ]
+
+  @spec create(t(), (DbNotification.t() -> t())) :: t()
+  def create(notification_without_id, insertion_callback \\ fn _ -> nil end) do
```

Comment:
This callback seems not very elixir-y to me. What about either doing the log call inside this function, or having the caller case on whether they got an id back or not?

Reply:
Logging within this function does make sense to me, because we've established that it's useful to log the creation of every single notification. But I do like the approach of passing a callback better than examining the result in the caller. It's more explicit IMHO to say "here is a way to request some processing be done if we successfully insert", than to have to know that a nil ID means insert failed in the caller.